torchtune/utils/generation.py

```diff
@@ -163,8 +163,9 @@ def generate(
         eos_reached = torch.zeros(batch_size, dtype=torch.bool, device=device)
         for cur_pos in range(min_prompt_len, total_gen_len):
             input_ids = tokens[:, prev_pos:cur_pos]
+            input_pos = torch.arange(prev_pos, cur_pos)
             if incremental_decode:
-                outputs = self.decoder_lm(input_ids, curr_pos=prev_pos)
+                outputs = self.decoder_lm(input_ids, input_pos=input_pos)
             else:
                 outputs = self.decoder_lm(input_ids)
             if logits_accessor:
```

Comment:
I know it's not part of these changes, but I find the name "accessor" a bit confusing tbh. Is the idea here to do some sort of slicing along the vocab dim or something? What is the difference between this and logits_transforms? Is it just that one is applied before softmax and the other is applied after?

Reply:
OK coming back to this now I'm a bit confused (prob should have read the full recipe code first). Why isn't this file deleted entirely? Seems you've covered all the generation logic in the recipe already. Or am I missing something?

Reply:
No particular reason, was leaving this as a follow up. Let me think about this - maybe I should combine the change here